### ISSUE TITLE
eigs() throwing unhelpful errors when max dim is reached.

### ISSUE THREAD

**Issue by nickhale** [chebfun/chebfun#1483]
Consider the following:

```
pref = cheboppref();
pref.minDimension = 9;
pref.maxDimension = 18;
A = chebop([-1,1]);
A.op = @(u) .001*diff(u,2) + u;
A.bc = 'dirichlet';
```

The first eigenmode requires 62 points to resolve (not shown here - take my word for it) so maxDimension = 25 will not be enough. This is what happens when we call eigs:
```
>> [V, D] = eigs(A, pref);
Error using mat2cell (line 97)
Input arguments, D1 through D2, must sum to each dimension of the input matrix size, [9  1].'
Error in chebDiscretization/mat2fun (line 44)
values = mat2cell( values, componentLength, size(values,2) );

Error in linop/eigs (line 278)
    u = mat2fun(discA, P*V);

Error in chebop/eigs (line 109)
[varargout{1:nargout}] = eigs(L, varargin{:});
```
The problem is that the discretization length gets incorrectly updated.

Perhaps more worryingly:
```
>> D = eigs(A, 1, pref)
D =
   0.960312668471022
```
if we simply ask for the eigenvectors we get a result with no warning (even though we know these values haven't converged!).

**Comment by nickhale** [comment]
In `feature-eigs-warning` I've modified the `eigs()` code so that the above now returns:
```
>> [V, D] = eigs(A, pref);
Warning: Maximimum dimension reached. Solution may not have converged. 
> In linop/eigs (line 255)
  In chebop/eigs (line 109) 
V =
   chebfun column (1 smooth piece)
       interval       length     endpoint values  
[      -1,       1]       17      0.12     0.12 
epslevel = 1.000004e-10   vscale = 1.283868e+00
D =
   0.685914643216041
```
and
```
>> D = eigs(A, 1, pref)
Warning: Maximimum dimension reached. Solution may not have converged. 
> In linop/eigs (line 255)
  In chebop/eigs (line 109) 
D =
   0.685914643216041
```

We can easily change the warning to be a (sensible) error, if that is preferred.

**Comment by asgeirbirkis** [comment]
@nickhale for the warning, should we point users to `cheboppref.maxDimensions`?

**Comment by nickhale** [comment]
Sounds good.

**Comment by asgeirbirkis** [comment]
Closed via #1484.

**Comment by JamesMathews** [comment]
Hi

When calculated the null space of a second order differential equation, I got the error:
```
Error using mat2cell (line 97)
Input arguments, D1 through D2, must sum to each dimension of the input matrix size, [2896   440].'
Error in opDiscretization/mat2fun (line 44)
values = mat2cell( values, componentLength, size(values,2) );
```

There is no reference to what is actually going wrong with Chebfun, but it looks to be a similar error to this. I am using Chebfun v5.3. So it looks like this merge didn't happen.
